pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -496,7 +479,7 @@ func (facts *PipelineRunFacts) GetPipelineTaskStatus() map[string]string {
 			case t.isSuccessful():
 				s = v1beta1.TaskRunReasonSuccessful.String()
 			// execution status is Failed when a task has succeeded condition with status set to false
-			case t.isConditionStatusFalse():
+			case (t.IsCustomTask() && t.areRunObjectsConditionStatusFalse()) || (!t.IsCustomTask() && t.areTaskRunsConditionStatusFalse()):
```

Comment:
it was helpful when this was defined as a function on its own, this change makes it harder to parse and took a minute to compare with the one in 500 (which should be exactly the same)

Reply:
Okay, broke it out into separate functions. Wasn't sure of the naming since Lee mentioned the original was confusing but we already have `isFailure() and `isConditionStatusFalse()` refers to multiple TaskRuns.